### ISSUE TITLE
feature request: hermetic environments

### ISSUE THREAD

**Issue by tamird** [mitsuhiko/rye#735]
Suppose I have workspace members `A` and `B` with disjoint dependencies and I accidentally `import` a dependency of `B` in `A`; today rye will allow this silently, but it should not (because it will fail when `A` is deployed standalone).

**Comment by mitsuhiko** [comment]
Yes. One idea I had was to set up a custom import hook (at least optionally) in Rye to experiment if this can be prevented. I also want to experiment with multi versioning later down the line so that import hook would be needed anyways.
